custom_components/unifiprotect/switch.py

```diff
@@ -118,67 +126,73 @@ class UnifiProtectSwitch(UnifiProtectEntity, SwitchEntity):
+        self.device: Camera | Light = device
+        self._attr_name = f"{self.device.name} {self.entity_description.name}"
         self._switch_type = self.entity_description.key
 
+        if not isinstance(self.device, Camera):
+            return
+
+        if self.device.is_privacy_on:
+            self._previous_mic_level = 100
+            self._previous_record_mode = RecordingMode.ALWAYS
+        else:
+            self._previous_mic_level = self.device.mic_volume
+            self._previous_record_mode = self.device.recording_settings.mode
+
     @property
     def is_on(self) -> bool:
         """Return true if device is on."""
-        ufp_value = self._device_data[self.entity_description.ufp_value]
+        ufp_value = get_nested_attr(self.device, self.entity_description.ufp_value)
         if self._switch_type == _KEY_HIGH_FPS:
-            return ufp_value == TYPE_HIGH_FPS_ON
+            return ufp_value == VideoMode.HIGH_FPS
         return ufp_value is True
 
     async def async_turn_on(self, **kwargs):
         """Turn the device on."""
+
+        if self._switch_type == _KEY_STATUS_LIGHT:
+            _LOGGER.debug("Changing Status Light to On")
+            await self.device.set_status_light(True)
+
+        if isinstance(self.device, Light):
+            return
+
         if self._switch_type == _KEY_HDR_MODE:
             _LOGGER.debug("Turning on HDR mode")
-            await self.upv_object.set_camera_hdr_mode(self._device_id, True)
+            await self.device.set_hdr(True)
         elif self._switch_type == _KEY_HIGH_FPS:
             _LOGGER.debug("Turning on High FPS mode")
-            await self.upv_object.set_camera_video_mode_highfps(self._device_id, True)
+            await self.device.set_video_mode(VideoMode.HIGH_FPS)
         elif self._switch_type == _KEY_PRIVACY_MODE:
-            _LOGGER.debug("Turning Privacy Mode on for %s", self._device_data["name"])
-            self._device_data["save_mic_level"] = self._device_data["mic_volume"]
-            self._device_data["save_rec_mode"] = self._device_data["recording_mode"]
-            await self.upv_object.set_privacy_mode(
-                self._device_id, True, 0, TYPE_RECORD_NEVER
-            )
-        else:
-            _LOGGER.debug("Changing Status Light to On")
-            await self.upv_object.set_device_status_light(
-                self._device_id, True, self._device_type
-            )
-        await self.protect_data.async_refresh(force_camera_update=True)
+            _LOGGER.debug("Turning Privacy Mode on for %s", self.device.name)
+            self._previous_mic_level = self.device.mic_volume
+            self._previous_record_mode = self.device.recording_settings.mode
+            await self.device.set_privacy(True, 0, RecordingMode.NEVER)
 
     async def async_turn_off(self, **kwargs):
         """Turn the device off."""
+
+        if isinstance(self.device, Camera):
```

Comment:
Could save some indent by checking `_KEY_STATUS_LIGHT` first and then reversing this conditional and returning